File: server/CoolPizza.Core/Entities/Products/Product.cs
using CoolPizza.Core.Enums;

namespace CoolPizza.Core.Entities.Products;

public class Product: Entity
{
    public const int NameMaxLength = 128;
    
    public string Name { get; private set; }
    public string Description { get; private set; }
    public string BaseImg { get; private set; }
    public string BasePrice { get; private set; }
    public ProductType Type { get; private set; }
    public Guid CategoryId { get; private set;}
    public ICollection<Goods> Goods { get; private set; }
    public ICollection<Pizza> Pizzas { get; private set; }
}

File: client/src/store/types/Product.ts
import {UUID} from "./shared";
import {Ingredient} from "./Ingredient";
import {Category} from "./Category";

export interface Goods {
  id: UUID,
  imgUrl: string;
  price: number;
  details: string;
  weight: number;
  product_id: string;
}

export interface Pizza {
  id: UUID,
  imgUrl: string;
  price: number;
  size: number;
  dough: string;
  weight: number;
  product_id: UUID;
  ingredients: Ingredient[];
}

export enum ProductTypes {
  goods = "Goods",
  pizza = "Pizza"
}

export interface Product {
  id: UUID;
  name: string;
  description: string;
  baseImg: string;
  basePrice: number;
  type?: ProductTypes;
  categoryId: string;
  variations: Goods[] | Pizza[]
}

export interface GroupedProduct {
  category: Category;
  products: Product[];
}